pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -120,41 +120,56 @@ func SetDefaults_KubeProxyConfiguration(obj *KubeProxyConfiguration) {
 }
 
 func SetDefaults_KubeSchedulerConfiguration(obj *KubeSchedulerConfiguration) {
-	if obj.Port == 0 {
-		obj.Port = ports.SchedulerPort
+	if len(obj.SchedulerName) == 0 {
+		obj.SchedulerName = api.DefaultSchedulerName
 	}
-	if obj.Address == "" {
-		obj.Address = "0.0.0.0"
+
+	if obj.HardPodAffinitySymmetricWeight == 0 {
+		obj.HardPodAffinitySymmetricWeight = api.DefaultHardPodAffinitySymmetricWeight
 	}
-	if obj.AlgorithmProvider == "" {
-		obj.AlgorithmProvider = "DefaultProvider"
+
+	if obj.AlgorithmSource.Policy == nil &&
+		(obj.AlgorithmSource.Provider == nil || len(*obj.AlgorithmSource.Provider) == 0) {
+		val := SchedulerDefaultProviderName
+		obj.AlgorithmSource.Provider = &val
 	}
-	if obj.ContentType == "" {
-		obj.ContentType = "application/vnd.kubernetes.protobuf"
+
+	if policy := obj.AlgorithmSource.Policy; policy != nil {
+		if policy.ConfigMap != nil && len(policy.ConfigMap.Namespace) == 0 {
+			obj.AlgorithmSource.Policy.ConfigMap.Namespace = api.NamespaceSystem
+		}
 	}
-	if obj.KubeAPIQPS == 0 {
-		obj.KubeAPIQPS = 50.0
+
+	if len(obj.HealthzBindAddress) == 0 {
+		obj.HealthzBindAddress = fmt.Sprintf("0.0.0.0:%v", ports.SchedulerPort)
+	} else if !strings.Contains(obj.HealthzBindAddress, ":") {
```

Comment:
This check and similar ones below are going to cause IPv6 compatibility issues. Please use [go 'net' package](https://golang.org/pkg/net/) to add/find port in an address.

Reply:
Done, thanks for the tip. Much cleaner, too.